waitUntilBomProcessingComplete=false seems not working as documented

The cmd I was trying is
`mvn io.github.pmckeown:dependency-track-maven-plugin:upload-bom -Ddependency-track.waitUntilBomProcessingComplete=false -Ddependency-track.projectName=test -Ddependency-track.dependencyTrackBaseUrl=http://localhost:8080 -Ddependency-track.apiKey=test -Ddependency-track.projectVersion=latest`

The logging showed it's still waiting for processing 
`
[INFO] --- dependency-track-maven-plugin:1.1.3:upload-bom (default-cli) @ test ---
[INFO] Project Name: test
[INFO] Project Version: latest
[INFO] Plugin is configured to wait for BOM processing to complete
[INFO] BOM uploaded to Dependency Track server
[INFO] Checking for BOM analysis completion
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[INFO] Still processing: true
[ERROR] Polling for processing completion was interrupted so continuing: Call 'io.github.pmckeown.dependencytrack.upload.UploadBomAction$$Lambda$53/1678889928@6fd5717c' failed after 20 tries!
`

I looked into the source code and did not found any reference of "waitUntilBomProcessingComplete" property, is there anything I missed?

--- a/README.md
+++ b/README.md
@@ -30,8 +30,9 @@ in the `pluginManagement` section of your POM to avoid repetition.
 </pluginManagement>
 ```
 
-**IMPORTANT** Recent releases of Dependency Track include multiple components, so you must ensure that you target
-the api server component URL and not the front-end component URL.
+**IMPORTANT** Dependency Track includes a front-end and an api-server component on different ports (defaulting to
+8080 and 8081 respectively). You must ensure that you target the api server component (8081) and not the front-end
+component URL in the `dependencyTrackBaseUrl` property.
 
 #### Polling Configuration
 The plugin is configured to poll for results from the Dependency-Track server by default.  The polling configuration
@@ -55,6 +56,44 @@ can be changed by modifying the `pollingConfig` element in the plugin configurat
     </pollingConfig>
 </configuration>
 ```
+
+To set these options on the Command Line, you need to set the PollingConfig in your POM file to be set by a property
+when running a command.  
+
+Note that when doing this, **the defaults are change to the primitive defaults**, so polling is not enabled unless you
+provide these properties.
+
+**XML Configuration Example**
+```xml
+<project>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.github.pmckeown</groupId>
+                <artifactId>dependency-track-maven-plugin</artifactId>
+                <version>${dependency-track-maven-plugin.version}</version>
+                <inherited>false</inherited>
+                <configuration>
+                    <dependencyTrackBaseUrl>${env.DEPENDENCY_TRACK_BASE_URL}</dependencyTrackBaseUrl>
+                    <apiKey>${env.DEPENDENCY_TRACK_API_KEY}</apiKey>
+                    <pollingConfig>
+                        <enabled>${polling.enabled}</enabled>
+                        <attempts>${polling.attempts}</attempts>
+                        <pause>${polling.pause}</pause>
+                        <timeUnit>${polling.timeUnit}</timeUnit>
+                    </pollingConfig>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>
+```
+
+Then in your `mvn` command, you can override the default values for polling like this:
+```shell script
+mvn dependency-track -Dpolling.enabled=false -Dpolling.attempts=100 -Dpolling.pause=2 -Dpolling.timeUnit=MILLIS
+``` 
+
 
 
 #### Dependency-Track Configuration
@@ -121,14 +160,14 @@ first in your POM to ensure that it runs first.
 
 #### Configuration
 
-By default this goal will poll the Dependency-Track server after a BOM upload to ensure that it has been analysed 
-before continuing with plugin execution.  This can be disabled by setting the `waitUntilBomProcessingComplete` 
-parameter to `false`.
+By default, the plugin will poll the Dependency-Track server after a BOM upload to ensure it has been completed before 
+continuing with plugin execution. This can be disabled by updating the [Polling Configuration](#polling-configuration). 
+
+The BOM Location can be modified if you generate it to a folder that is not the default of `target/`.
 
 |Property                      |Required|Default Value |
 |------------------------------|--------|--------------|
 |bomLocation                   |false   |target/bom.xml|
-|waitUntilBomProcessingComplete|false   |true          |
 
 ### Get Project Findings
 After a BOM upload, the best way to determine if there are any vulnerabilities is to use the `findings` goal which is 

--- a/src/main/java/io/github/pmckeown/dependencytrack/PollingConfig.java
+++ b/src/main/java/io/github/pmckeown/dependencytrack/PollingConfig.java
@@ -1,5 +1,8 @@
 package io.github.pmckeown.dependencytrack;
 
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
 import javax.inject.Singleton;
 import java.time.temporal.ChronoUnit;
 
@@ -63,4 +66,9 @@ public class PollingConfig {
     }
 
     public enum TimeUnit {SECONDS, MILLIS}
+
+    @Override
+    public String toString() {
+        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
+    }
 }

--- a/src/main/java/io/github/pmckeown/dependencytrack/upload/UploadBomAction.java
+++ b/src/main/java/io/github/pmckeown/dependencytrack/upload/UploadBomAction.java
@@ -37,8 +37,7 @@ public class UploadBomAction {
     public boolean upload(String bomLocation) throws DependencyTrackException {
         logger.info("Project Name: %s", commonConfig.getProjectName());
         logger.info("Project Version: %s", commonConfig.getProjectVersion());
-        logger.info("Plugin %s configured to wait for BOM processing to complete",
-                commonConfig.getPollingConfig().isEnabled() ? "is" : "is not");
+        logger.info("%s", commonConfig.getPollingConfig());
 
         Optional<String> encodedBomOptional = bomEncoder.encodeBom(bomLocation, logger);
         if (!encodedBomOptional.isPresent()) {

